test/integration/stats_integration_test.cc

```diff
@@ -259,7 +259,8 @@ TEST_P(ClusterMemoryTestRunner, MemoryLargeClusterSizeWithFakeSymbolTable) {
   // 2019/10/17  8484     43340       44000   stats: add unit support to histogram
   // 2019/11/01  8859     43563       44000   build: switch to libc++ by default
   // 2019/11/15  9040     43371       44000   build: update protobuf to 3.10.1
-  // 2019/11/15  9040     43403       35500   upstream: track whether cluster is local
+  // 2019/11/15  9031     43403       44000   upstream: track whether cluster is local
+  // 2019/12/04  9227     49531       50000   router: per-cluster histograms w/ timeout budget
```

Comment:
I think we should make these histograms opt-in similar to the way the circuit_breaker.proto has a flag for extra stats:
https://github.com/envoyproxy/envoy/blob/master/api/envoy/api/v2/cluster/circuit_breaker.proto#L49

It would cut down on the default memory usage and I'd argue this stat isn't for everyone. See if you can mimic something like this below?
https://github.com/envoyproxy/envoy/blob/81d3bebf9187f408c829dde367214069651a3768/source/common/upstream/upstream_impl.cc#L1043